hal/src/rtl872x/usart_hal.cpp

```diff
@@ -572,6 +574,7 @@ class Usart {
                 } else {
                     UART_TXDMACmd(uartInstance, ENABLE);
                     BaseType_t yield = pdFALSE;
+                    uart->busy_ = false;
```

Comment:
Should probably check `transmitting_` here just in case to avoid a weird race condition between starting a new DMA transfer and finishing a previous one based on TX FIFO state.

Reply:
`enableEvent()` should also be patched probably to clear the tx bit before enabling the interrupt.
```c++
        if (event & HAL_USART_PVT_EVENT_WRITABLE) {
            if (space() <= 0) {
                TxLock lk(this);
                xEventGroupClearBits(evGroup_, HAL_USART_PVT_EVENT_WRITABLE);
                UART_INTConfig(uartInstance, RUART_IER_ETBEI, ENABLE);
                // Temporarily disable TX DMA, otherwise, the TX FIFO won't be empty until all data is transferred by DMA.
                UART_TXDMACmd(uartInstance, DISABLE);
                if (!transmitting_) {
                    startTransmission();
                }
            } else {
                xEventGroupSetBits(evGroup_, HAL_USART_PVT_EVENT_WRITABLE);
            }
        }
```